recipes/osgearth/all/conanfile.py

```diff
@@ -23,192 +25,166 @@ class OsgearthConan(ConanFile):
-
-        if self.settings.compiler == "gcc" and self.settings.compiler.version == "11":
-            # need draco >= 1.4.0 for gcc11
-            # https://github.com/google/draco/issues/635
-            self.options.with_draco = False
+            self.options.rm_safe("fPIC")
 
     def requirements(self):
-
         self.requires("opengl/system")
-        self.requires("gdal/3.4.3")
-        self.requires("openscenegraph/3.6.5")
-        self.requires("libcurl/8.2.0")
-        self.requires("lerc/2.2")
+        self.requires("gdal/3.10.3")
+        self.requires("openscenegraph/3.6.5", transitive_headers=True)
+        self.requires("libcurl/[>=7.78.0 <9]")
+        self.requires("lerc/4.0.1")
         self.requires("rapidjson/1.1.0")
+        self.requires("sqlite3/[>=3.42 <4]")
 
-        self.requires("zlib/[>=1.2.11 <2]")  # override
-        self.requires("libtiff/4.5.1")  # override
-        self.requires("libpng/1.6.40")  # override
-
-        # if self.options.build_triton_nodekit:
-        #     self.requires("triton_nodekit")
-        # if self.options.build_silverlining_nodekit:
-        #     self.requires("silverlining_nodekit")
-        if self.options.build_leveldb_cache:
-            self.requires("leveldb/1.22")
         if self.options.build_rocksdb_cache:
             self.requires("rocksdb/6.29.5")
         if self.options.build_zip_plugin:
-            self.requires("zstd/1.5.5")  # override
+            self.requires("libzip/1.7.3")
         if self.options.with_geos:
-            self.requires("geos/3.11.1")
-        if self.options.with_sqlite3:
-            self.requires("sqlite3/[>=3.42 <4]")
-        if self.options.with_draco:
-            self.requires("draco/1.4.3")
-        # if self.options.with_basisu:
-        #     self.requires("basisu")
-        # if self.options.with_glew:
-        #     self.requires("glew/2.2.0")
+            self.requires("geos/3.12.0")
         if self.options.with_protobuf:
-            self.requires("protobuf/3.21.9")
+            self.requires("protobuf/3.21.12")
         if self.options.with_webp:
-            self.requires("libwebp/1.3.1")
-
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
-        for package in ("Draco", "GEOS", "LevelDB", "OSG", "RocksDB", "Sqlite3", "WEBP"):
-            # Prefer conan's find package scripts over osgEarth's
-            os.unlink(os.path.join(self._source_subfolder, "CMakeModules", "Find{}.cmake".format(package)))
+            self.requires("libwebp/[>=1.3.2 <2]")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-        self._patch_sources()
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        apply_conandata_patches(self)
+
+    def _get_library_postfix(self, build_type: str) -> str:
```

Comment:
Hi @tylerbrawl - coming back to this, the only remaining doubt about this recipe for me now is this function.

Could you share some more info about this? Correct me if I'm missing something, but the idea is that the dependency openscenegraph expects to load plugins with a given postfix when using it, with no way to override the behaviour, and that we then need to create our libraries with said postfixes?

I wanted to know more because we usually try to refrain from modifying the library names that recipes generate unless absolutely necessary, so I'm wondering if there could be simpler/different approaches :)

Reply:
Yes, I think raising this with upstream would be worth it if only to let them know.

Do note that other package managers that have this library seem not to care about the output name of the plugins, so I wonder then how they are making them work, if at all

Reply:
@AbrilRBS , see how this logic was already present in the original recipe:
https://github.com/conan-io/conan-center-index/pull/28101/files#diff-33d8c7904e00023d56217e34e179f3e7aeaf424518888d0259fbf3e4089770bbL204

It is possible that other plugins do not have it in place and the problem went unnoticed, as they would still load fine when their build_type is `release`.